docs/src/man/comparisons.md

```diff
@@ -235,6 +235,66 @@ The table below compares more advanced commands:
+| Pick & transform columns           | `df[, list(mean(x), y)]`                         | `select(df, :x => mean, :y)`                 |
+| Pick rows                          | `df[ x >= 1 ]`                                   | `filter(:x => >=(1), df)`                    |
+| Sort rows (in place)               | `setorder(df, x)`                                | `sort!(df, :x)`                              |
+| Sort rows                          | `df[ order(x) ]`                                 | `sort(df, :x)`                               |
+
+### Grouping data and aggregation
+
+| Operation                   | data.table                                        | DataFrames.jl                             |
+|:----------------------------|:--------------------------------------------------|:------------------------------------------|
+| Reduce multiple values      | `df[, list(mean(x)), by = list(id) ]`             | `combine(groupby(df, :id), :x => mean)`   |
+| Add new columns (in place)  | `df[, x_mean := mean(x), by = list(id) ]`         | `transform!(groupby(df, :id), :x => mean)`|
+| Pick & transform columns    | `df[, list(x_mean = mean(x), y), by = list(id) ]` | `select(groupby(df, :id), :x => mean, :y)`|
+
+### More advanced commands
+
+| Operation                         | data.table                                                                                 | DataFrames.jl                                                               |
+|:----------------------------------|:-------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------|
+| Complex Function                  | `df[, list(mean(x, na.rm = T)) ]`                                                          | `combine(df, :x => x -> mean(skipmissing(x)))`                              |
+| Transform certain rows (in place) | `df[x<=0, x:=0 ]`                                                                          | `df.x[df.x .<= 0] .= 0`                                                     |
+| Transform several columns         | `df[, list(max(x), min(y)) ]`                                                              | `combine(df, :x => maximum, :y => minimum)`                                 |
+|                                   | `df[, lapply(.SD, mean), .SDcols = c("x", "y") ]`                                          | `combine(df, [:x, :y] .=> mean)`                                            |
+|                                   | `df[, lapply(.SD, mean), .SDcols = patterns("x*") ]`                                       | `combine(df, names(df, r"^x") .=> mean)`                                    |
+|                                   | `df[, unlist(lapply(.SD, function(x) c(max=max(x), min=min(x)))), .SDcols = c("x", "y") ]` | `combine(df, ([:x, :y] .=> [maximum minimum])...)`                          |
```

Comment:
Looks like we have some work to do on this one. I can't think of an easier way right now. There may be an outstanding issue or pull request, maybe @jangorecki @MichaelChirico recall. I never wanted to encourage wide data, so my focus was on long. But I know people like to go wide like this, perhaps for presenting results in a paper or web page, so this task should be easier.

Reply:
It's cheating a bit &mdash; or maybe not &mdash; but I'd probably use `dcast` here.

```r
dcast(df, .~., fun=list(min, max), value.var = c('x', 'y'))
```

The advantage of this approach is that it also scales well to cases where you want to collapse by group. I think the 'unlist' approach would struggle here.

```r
dcast(df, grp~., fun=list(min, max), value.var = c('x', 'y'))
```

Mind you, grouping is something that the DataFrames.jl implementation automatically supports (and, to @mattdowle's point, might be conceptually simpler my `dcast` workflow).

```julia
combine(groupby(df, :grp), ([:x, :y] .=> [minimum maximum])...)
```

Reply:
I would do `df[, c(lapply(.SD, max), lapply(.SD, min)), .SDcols = c("x", "y")]`. That _should_ `GForce` as well where the `unlist` one will not.

Reply:
I guess it doesn't use GForce, and also, it results in duplicate names! ouch